src/components/rangeFilterInput/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC } from 'react';
+import {
+  InputGroup,
+  NumberInput,
+  NumberInputField,
+  NumberInputProps,
+} from '@chakra-ui/react';
+
+import Stack from '../stack';
+import InputLeftElement from './InputLeftElement';
+
+type InputElement = 'from' | 'to';
+
+type ChangeCallback = {
+  touched: InputElement;
+  value: {
+    from: number;
+    to: number;
+  };
+};
+
+type RangeFilterInputProps = {
+  name: {
+    from: string;
+    to: string;
+  };
+  handleChange: (event: ChangeCallback) => void;
+  placeholder?: {
+    from: string;
+    to: string;
+  };
+  unit?: string;
+  value: {
+    from: number;
+    to: number;
+  };
+  size?: 'md' | 'lg';
+} & Pick<NumberInputProps, 'min' | 'max' | 'isDisabled'>;
+
+const RangeFilterInput: FC<RangeFilterInputProps> = ({
+  name,
+  handleChange,
+  placeholder,
+  unit,
+  value,
+  size = 'md',
+  ...rest
+}) => {
+  return (
+    <Stack direction="row" spacing={0}>
+      {Object.keys(name).map((elem: string) => {
+        const inputFieldElement = elem as InputElement;
+
+        return (
+          <InputGroup key={inputFieldElement}>
+            <NumberInput
+              width="full"
+              variant={`${inputFieldElement}Outline`}
+              defaultValue={value ? value[inputFieldElement] : ''}
+              size={size}
+              name={name[inputFieldElement]}
+              onChange={(val) =>
+                handleChange({
+                  touched: inputFieldElement,
+                  value: { ...value, [inputFieldElement]: val },
+                })
+              }
+              {...rest}
+            >
+              {unit ? <InputLeftElement unit={unit} /> : null}
+              <NumberInputField
+                value={value ? value[inputFieldElement] : ''}
+                placeholder={placeholder ? placeholder[inputFieldElement] : ''}
+              />
+            </NumberInput>
+          </InputGroup>
+        );
+      })}
```

Comment:
I wouldn't iterate here. You can extract the individual input as a separate sub-component and create two of them (`from` and `to`). This would make the code more readable and eliminate the need for the type casting.
Also if you'd change the interface to `from` and `to` you could just spread them here nicely

Reply:
Yea initially I did it like that, then I look into carforyou implementation and refactor it to this. But I would also prefer it like that. Will adjust it with new prop structure.